src/components/SwapWidget/LimitOrder/index.tsx

```diff
@@ -163,6 +177,54 @@ const LimitOrder: React.FC<Props> = ({ swapType, setSwapType, isLimitOrderVisibl
     [onUserInput],
   );
 
+  // setting default tokens
+  const filterCurrencies = (symbolName: string): Currency[] => {
+    const tokens = Object.values(allTokens);
+    tokens.unshift(CAVAX[chainId] as Token);
+    const _tokens = tokens.filter((token) => token !== CAVAX[chainId]);
+    const currencies = CHAINS[chainId]?.evm ? [CAVAX[chainId], ..._tokens] : _tokens;
+    return currencies.filter((token) => token?.symbol?.toLowerCase() === symbolName.toLowerCase());
+  };
+
+  const setDefaultCurrency = (tokenDrawerType: LimitNewField, currency: Currency) => {
+    if (tokenDrawerType === (LimitNewField.INPUT as any)) {
+      setApprovalSubmitted(false); // reset 2 step UI for approvals
+    }
+
+    // here need to add isToken because in Galato hook require this variable to select currency
+    const newCurrency: any = { ...currency };
+    if (currency?.symbol === CAVAX[chainId].symbol) {
+      newCurrency.isNative = true;
+    } else {
+      newCurrency.isToken = true;
+    }
+
+    onCurrencySelection(tokenDrawerType as any, newCurrency);
+    // this is to update tokens on chart on token selection
+    onSwapCurrencySelection(tokenDrawerType as any, currency);
+  };
```

Comment:
`setDefaultCurrency` logic is same as `onCurrencySelect`, only difference is one new argument, try to reuse the code